packages/expo-updates/scripts/create-manifest-ios.sh

```diff
@@ -21,8 +21,8 @@ NODE_BINARY=${NODE_BINARY:-node}
 
 # ref: https://github.com/facebook/react-native/blob/c974cbff04a8d90ac0f856dbada3fc5a75c75b49/scripts/react-native-xcode.sh#L59-L65
 EXPO_UPDATES_PACKAGE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
-# The project should be located next to where expo-updates is installed
-# in node_modules.
+# If PROJECT_ROOT is not specified, fallback to use Xcode PROJECT_DIR
+PROJECT_ROOT=${PROJECT_ROOT:-"$PROJECT_DIR"}
```

Comment:
does `$PROJECT_DIR` evaluate to the ios directory? If so, I think this should fall back to `$PROJECT_DIR/..`.

Reply:
actually `$PROJECT_DIR` is something like `/project/ios/Pods`.

there is `..` in the following line and map to `/project/ios`:
https://github.com/expo/expo/blob/7ef5b4b99a2b04666ad0631f378f15acaa11009e/packages/expo-updates/scripts/create-manifest-ios.sh#L43

in `createManifest.js`, there is yet another `..` and map to `/project`:
https://github.com/expo/expo/blob/7ef5b4b99a2b04666ad0631f378f15acaa11009e/packages/expo-updates/scripts/createManifest.js#L19-L21

the logic is pretty the same to expo-constants 😅  let me know for any comments, i could update these two modules together.

Reply:
the code is really confused as you mentioned. https://github.com/expo/expo/pull/14198/commits/c18486b8e213621b59768bda39b5ebbb9ec6f9c5 tries to make it clearer.